src/win/pipe.c

```diff
@@ -942,6 +956,89 @@ int uv_pipe_listen(uv_pipe_t* handle, int backlog, uv_connection_cb cb) {
 }
 
 
+#ifdef UV_USE_PIPE_INTERRUPTER
+static DWORD WINAPI uv__readfile_interrupter_thread(void* param) {
+  uv__readfile_interrupter_t* interrupter;
+  int r;
+
+  interrupter = param;
+  for (;;) {
+    r = WaitForSingleObject(interrupter->completed_event,
```

Comment:
This doesn't need to be run in a dedicated long-poll thread, you can call `RegisterWaitForSingleObject` instead.